src/widgets/modalChildren/ADRInput.js

```diff
@@ -1,30 +1,94 @@
 /* eslint-disable react/forbid-prop-types */
-import React, { useState } from 'react';
-import { StyleSheet } from 'react-native';
+import React, { useEffect, useState, useLayoutEffect } from 'react';
+import { ActivityIndicator, StyleSheet, Text } from 'react-native';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import moment from 'moment';
 import { JSONForm } from '../JSONForm/JSONForm';
 import { UIDatabase } from '../../database/index';
 import { FlexRow } from '../FlexRow';
+import { FlexView } from '../FlexView';
 import { PageButton } from '../PageButton';
-import { SUSSOL_ORANGE } from '../../globalStyles/colors';
+import { SUSSOL_ORANGE, WHITE } from '../../globalStyles/colors';
 import globalStyles from '../../globalStyles/index';
 import { PatientActions } from '../../actions/PatientActions';
 import { selectCurrentPatient } from '../../selectors/patient';
+import { useLocalAndRemotePatientHistory } from '../../hooks/useLocalAndRemoteHistory';
+import { dispensingStrings, generalStrings, vaccineStrings } from '../../localization';
 
-export const ADRInputComponent = ({ onCancel, onSave, patient }) => {
+const getSchemaItems = jsonSchema => {
+  const { properties = {} } = jsonSchema;
+  const { causes = {} } = properties;
+  const { items } = causes;
+  return items;
+};
+const mapHistory = history =>
+  history.map((h, index) => {
+    const { prescriberOrVaccinator, itemName, confirmDate } = h;
+    const date = moment(confirmDate).format('DD/MM/YY');
+    const vaccinator = prescriberOrVaccinator
+      ? `  ${vaccineStrings.vaccinator}: ${prescriberOrVaccinator}`
+      : '';
+
+    return `${index + 1}. ${itemName}  ${generalStrings.date}: ${date}${vaccinator}`;
+  });
+
+const LoadingIndicator = ({ loading }) => (
+  <FlexView flex={1} justifyContent="center" alignItems="center" style={{ marginTop: 20 }}>
+    {loading && (
+      <>
+        <Text style={localStyles.text}>{dispensingStrings.fetching_history}</Text>
+        <ActivityIndicator color={SUSSOL_ORANGE} size="small" style={{ marginTop: 10 }} />
+      </>
+    )}
+  </FlexView>
+);
+
+LoadingIndicator.propTypes = {
+  loading: PropTypes.bool.isRequired,
+};
+
+export const ADRInputComponent = ({ onCancel, onSave, patient, patientHistory }) => {
```

Comment:
I thiiink the regular dispensing local history with all items is also being passed in here - might also need an update (`selectVaccinePatientHistory` vs `selectPatientHistory`)?

Reply:
another good catch! thank you :)
eafe967c7

I've gone with filtering, as I have the vague though that this might want to be expanded to include all patient history for some clients and thought that it was handy to have things in one location.